### PR TITLE
Clarify boothook docs

### PR DESCRIPTION
**Pull request by holmanb**
```
Clarify boothook docs
```

## Additional Context
See commit messages for explanations

### CODE REVIEW

**Review by s-makin — APPROVED** [review]
I think this makes sense. Thanks for taking the lead on this - I agree it's better not to present these terms without them being explained anywhere, and if the user doesn't need to know about boothooks at all then this definition/term should be avoided except in the dev docs.

**Review by aciba90 — APPROVED** [review]
LGTM, thanks for cleaning this up!

File: doc/rtd/explanation/boot.rst [hunk 1]
```diff
@@ -107,9 +107,7 @@ mounted, including ones that have stale (previous instance) references in
 :file:`/etc/fstab`. As such, entries in :file:`/etc/fstab` other than those
 necessary for cloud-init to run should not be done until after this stage.
 
-A part-handler will run at this stage, as will boothooks including
-cloud-config ``bootcmd``. The user of this functionality has to be aware
-that the system is in the process of booting when their code runs.
+A part-handler and boothooks will run at this stage.
```
Comment (blackboxsw) [inline doc/rtd/explanation/boot.rst:110]: Let's cross link using the autosectiondocs.
```suggestion
A part-handler and :ref:`boothooks<explanation/format:\`\`cloud-boothook\`\`>`
+will run at this stage.
```

**Review by blackboxsw — CHANGES_REQUESTED** [review]
MOAR linking FTW. Approved after this changeset.

**Review by blackboxsw — APPROVED** [review]
There was a stray `+` as part of my suggested fix. force-pushed that and +1

**Pull request merged**